File: src/backend/internal/model/attachment.go
package model

import "time"

// MessageAttachment 消息附件
type MessageAttachment struct {
	ID            string    `json:"id" db:"id"`
	MessageID     string    `json:"message_id" db:"message_id"`
	FileName      string    `json:"file_name" db:"file_name"`
	MimeType      string    `json:"mime_type" db:"mime_type"`
	FileSize      int64     `json:"file_size" db:"file_size"`
	FilePath      string    `json:"file_path" db:"file_path"`
	ThumbnailPath string    `json:"thumbnail_path,omitempty" db:"thumbnail_path"`
	Width         int       `json:"width,omitempty" db:"width"`
	Height        int       `json:"height,omitempty" db:"height"`
	CreatedAt     time.Time `json:"created_at" db:"created_at"`
}

File: src/frontend/src/types/attachment.ts
export interface AttachmentPayload {
  file_name: string;
  mime_type: string;
  file_size: number;
  file_path: string;
  thumbnail_path: string | null;
  width: number | null;
  height: number | null;
}

export interface MessageAttachment {
  id: string;
  message_id: string;
  file_name: string;
  mime_type: string;
  file_size: number;
  file_path: string;
  thumbnail_path: string | null;
  width: number | null;
  height: number | null;
  created_at: string;
}

export function isImageAttachment(mimeType: string): boolean {
  return mimeType.startsWith('image/');
}

export function isPDFAttachment(mimeType: string): boolean {
  return mimeType === 'application/pdf';
}

/**
 * 判断是否为 PowerPoint 演示文稿（.pptx / .ppt）。
 * mime 优先（pptx / ppt 的官方 mime），但上传链路 mime 可能不准，故用文件名后缀兜底。
 */
export function isPptxAttachment(mimeType: string, fileName: string): boolean {
  const ppMimes = new Set([
    'application/vnd.openxmlformats-officedocument.presentationml.presentation',
    'application/vnd.ms-powerpoint',
  ]);
  if (ppMimes.has(mimeType)) return true;
  return /\.pptx?$/i.test(fileName);
}

export function formatFileSize(bytes: number): string {
  if (bytes < 1024) return `${bytes} B`;
  if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`;
  return `${(bytes / (1024 * 1024)).toFixed(1)} MB`;
}
